### PR TITLE
BLO-27 Generified methods for param query filtering

### PR DESCRIPTION
**Pull request by Svenmarim**

### CODE REVIEW

File: src/repositories/concrete/mongoBlockchainRepository.ts [hunk 1]
```diff
@@ -16,84 +16,25 @@ export class MongoBlockchainRepository implements IBlockchainRepository {
...
-            const database = await this.client.connectAsync();
-            const collection = database.collection(this.tableName);
-
-            return new Promise((resolve, reject) => {
-                collection.findOne({ "blockHeader.blockNumber": blockNumber }).then((result) => {
-                    if (result) {
-                        resolve(result);
+            const allBlocks: Block[] = await database.collection(this.tableName);
+            let filteredBlocks: Block[] = new Array();
+            for (const block of allBlocks) {
```
Comment (saltz) [inline src/repositories/concrete/mongoBlockchainRepository.ts]: This for -> for -> if -> if code block can be simplified the following way:

```ts
for (const query of queries) {
    const field = blocks[0].blockHeader[query[0]];
                
    if (field) {
        throw FilterException("Field in query does not exists");
    }

    filteredBlocks.push.apply(filteredBlocks,
        (blocks.filter((block) => block.blockHeader[query[0]] === query[1])));
    }
}
```

File: src/repositories/concrete/mongoBlockchainRepository.ts [hunk 2]
```diff
@@ -122,4 +63,14 @@ export class MongoBlockchainRepository implements IBlockchainRepository {
             this.client.disconnectAsync();
         }
     }
+
+    private paramQueryEqualsToFieldName(queryName: string): boolean {
```
Comment (saltz) [inline src/repositories/concrete/mongoBlockchainRepository.ts]: This can be removed if the example above is implemented.

**Review by saltz — DISMISSED** [review]

File: src/repositories/concrete/mongoTransactionRepository.ts [hunk 3]
```diff
@@ -19,17 +20,20 @@ export class MongoTransactionRepository implements ITransactionRepository {
...
-            for (const transaction of allTransactions) {
-                for (const query of queries) {
-                    if (this.paramQueryEqualsToFieldName(query[0])) {
-                        if (transaction.query[0] === query[1]) {
-                            filteredTransactions.push(transaction);
-                        }
-                    }
+            
+            for (const query of queries) {
+                const field = allTransactions[0][query[0]];
```
Comment (quintaartsen) [inline src/repositories/concrete/mongoTransactionRepository.ts]: move the duplicate code to a separate class `RepositoryOperations` where these methods can live.

**Review by quintaartsen — CHANGES_REQUESTED** [review]

**Comment by saltz** [comment]
The build is still failing @Svenmarim, please fix this before we start our reviews.

**Review by quintaartsen — APPROVED** [review]

**Review by roelvdboom — APPROVED** [review]

**Pull request merged**